src/object.d

```diff
@@ -3722,6 +3722,133 @@ template _arrayOp(Args...)
     alias _arrayOp = arrayOp!Args;
 }
 
+/*
+ * Support for switch statements switching on strings.
+ * Params:
+ *      caseLabels = sorted array of strings generated by compiler. Note the
+                   strings are sorted by length first, and then lexicographically.
+ *      condition = string to look up in table
+ * Returns:
+ *      index of match in caseLabels, -1 if not found
+*/
+int __switch(T, caseLabels...)(T[] condition)
+{
+    // This closes recursion for other cases.
+    static if (caseLabels.length == 0)
+    {
+        return -1;
+    }
+    else static if (caseLabels.length == 1)
+    {
+        return __cmp(condition, caseLabels[0]) == 0 ? 0 : -1;
+    }
+    // To be adjusted after measurements
+    // Compile-time inlined binary search.
+    else static if (caseLabels.length < 7)
+    {
+        int r = void;
+        if (condition.length == caseLabels[$ / 2].length)
+        {
+            r = __cmp(condition, caseLabels[$ / 2]);
+            if (r == 0) return cast(int) caseLabels.length / 2;
+        }
+        else
+        {
+            // Equivalent to (but faster than) condition.length > caseLabels[$ / 2].length ? 1 : -1
+            r = ((condition.length > caseLabels[$ / 2].length) << 1) - 1;
+        }
+
+        if (r < 0)
+        {
+            // Search the left side
+            return __switch!(T, caseLabels[0 .. $ / 2])(condition);
+        }
+        else
+        {
+            // Search the right side
+            r = __switch!(T, caseLabels[$ / 2 + 1 .. $])(condition);
+            return r != -1 ? cast(int) (caseLabels.length / 2 + 1 + r) : -1;
+        }
+    }
+    else
+    {
+        // Run-time binary search in a static array of labels.
+        static const T[][caseLabels.length] cases =  [ caseLabels ];
+        size_t low = 0;
+        size_t high = cases.length;
+
+        do
+        {
+            auto mid = (low + high) / 2;
```

Comment:
Unfortunate that nobody saw this during the review, this binary search does not depend on the template arguments, but is duplicated in every instance of the `__switch` template.
In order to reuse template bloat, this code should be factored out into a non-templated function.